src/main/scala/chisel3/util/Valid.scala

```diff
@@ -34,7 +36,7 @@ object Valid {
 object Pipe
 {
   @chiselName
-  def apply[T <: Data](enqValid: Bool, enqBits: T, latency: Int): Valid[T] = {
+  def apply[T <: Data](enqValid: Bool, enqBits: T, latency: Int)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Valid[T] = {
```

Comment:
Source info shouldn't be propagated through helper functions, only chisel internals.
Source info support for libraries is something that needs more thought and discussion.